crates/core/tedge_agent/src/tedge_operation_converter/actor.rs

```diff
@@ -189,7 +189,7 @@ impl TedgeOperationConverterActor {
                     on_success,
                     on_error,
                 );
-                self.restart_sender.send(cmd).await?;
+                self.restart_sender.as_mut().unwrap().send(cmd).await?;
```

Comment:
Properly handling the case where there no `restart_sender` is a must, because disabling the feature will not prevent the agent from receiving restart requests.